### ISSUE TITLE
Support MSI MPG Z690 EDGE WIFI DDR4 (ALC4080)

### ISSUE THREAD

**Issue by tekstryder** [alsa-project/alsa-ucm-conf#210]
Per @perexg request, filing a new issue after testing git master branch changes for the ALC4080 and finding any profile(s) to be missing/nonfunctional.   

USB ID: `0db0:1feb`   

In version 1.2.7.2 of .../ucm2/USB-Audio/Realtek/ALC4080-HiFi.conf I need to change:

- Define.SPDIFIndex "3" to Define.SPDIFIndex "2"

For proper playback via S/PDIF. After that change the release version is working great as far as I could test.

Please let me know if I can provide any other useful information.

Attaching the output of `alsaucm -c hw:2 dump text`   
Attaching the output of `pactl list sinks`

[ucm-dump.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9463734/ucm-dump.txt)

[sinks.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9463784/sinks.txt)

**Comment by perexg** [comment]
I would like to see the ucm dump for the latest ucm configuration.

**Comment by tekstryder** [comment]
Yes, sorry, was just grabbing that after switching back to master. Opps.

[ucm-dump-master.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9463840/ucm-dump-master.txt)

**Comment by tekstryder** [comment]
Attached are dumps of alsa-info from both the working release version and ucm git master, along with the diff.

Not sure if the differences are noteworthy or glean any additional insight.

[alsa-info-release.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9470362/alsa-info-release.txt)
[alsa-info-master.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9470361/alsa-info-master.txt)
[alsa-info-diff-release-vs-master.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9470360/alsa-info-diff-release-vs-master.txt)

**Comment by tekstryder** [comment]
Yes I do:

`ALSA lib main.c:826:(execute_sequence) unable to execute cset 'name='PCM,1 Playback Switch' off'
ALSA lib main.c:2573:(set_verb_user) error: failed to initialize new use case: HiFi
alsaucm: error failed to set _verb=HiFi: No such file or directory`

No error when executing the same command on the release version.

**Comment by perexg** [comment]
Could you retest with the latest ucm configuration tree ? The above commit 593257c616fe47d46c7b8618f0fd0c4551938876 should fix this problem. Thank you.

**Comment by perexg** [comment]
We need the debug log from pulseaudio then - https://fedoraproject.org/wiki/How_to_debug_PulseAudio_problems .

**Comment by tekstryder** [comment]
> We need the debug log from pulseaudio then

Thanks. Attached is the verbose pulseaudio 16.1 startup log.

[pulse-debug-verbose.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9498838/pulse-debug-verbose.txt)

**Comment by tekstryder** [comment]
Oh, I found the issue, redundant hw:hw, lemme see if I can patch it properly...

**Comment by perexg** [comment]
Thanks, another two patches 439a85fa82006d22ebab2f54134b467d94f12494 and 89b151c7af5e437894cf820a34a4ab08752bc845 should resolve this problem. Could you test, please ?

**Comment by tekstryder** [comment]
Tested latest changes. S/PDIF output is working as expected now. Thank you!

For the sake of completeness this board also has front/rear mic inputs similar to the asus rog. I do not have a mic to test these ports unfortunately.

I also do not have any analog speakers to test any of the surround modes.

However. I was able to test both the front and rear headphone outputs and those are working correctly.
